tests/networking-hosts.nix

```
@@ -0,0 +1,35 @@
+{ config, pkgs, ... }:
+
+{
+  networking.hosts = {
+    "127.0.0.1" = [ "my.super.host" ];
+    "10.0.0.1" = [ "my.super.host" "my.other.host" ];
+  };
+
+  test = ''
+    echo checking /etc/hosts file >&2
+
+    file=${config.out}/etc/hosts
+
+    if [[ ! $(grep '127.0.0.1' $file | head -n1) =~ localhost$ ]]; then
+      cat $file
+      echo "'$(grep '127.0.0.1' $file | head -n1)'"
+      exit 1
+    fi
+    if [[ ! $(grep '127.0.0.1' $file | tail -n1) =~ my.super.host$ ]]; then
```

```suggestion
    grep '127.0.0.1' $file | tail -n1 | grep 'my.super.host$'
```

`set -e` should be enabled so if you can construct these as a single chain of commands then if statements shouldn't necessary

---

I verified and indeed if I meddle with the second grep, like replacing `host` with `hst`, the test fails.